src/node.rs

```diff
@@ -0,0 +1,293 @@
+            // Append entries to the Raft log
+            self.raw_node.mut_store().wl().append(&ready.entries).unwrap();
+        }
+
+        if let Some(ref hs) = ready.hs {
+            // Raft HardState changed, and we need to persist it.
+            self.raw_node.mut_store().wl().set_hardstate(hs.clone());
+        }
+
+        if !is_leader {
+            // We just stepped down as leader
+            if self.leader_state.is_some() {
+                // If we were building a block, cancel it
+                match self.leader_state {
+                    Some(LeaderState::Building(_)) => {
+                        debug!("Leader({}) stepped down, cancelling block", self.id);
+                        self.service.cancel_block().expect("Failed to cancel block");
+                    }
+                    _ => (),
+                }
+                self.leader_state = None;
+            }
+            // If not leader, the follower needs to reply the messages to
+            // the leader after appending Raft entries.
+            let msgs = ready.messages.drain(..);
+            for msg in msgs {
+                debug!("Peer({}) wants to send message to {}", self.id, msg.to);
+                self.send_msg(&msg);
+            }
+        }
+
+        if let Some(committed_entries) = ready.committed_entries.take() {
+            let mut _last_apply_index = 0;
+            for entry in committed_entries {
+                // Mostly, you need to save the last apply index to resume applying
+                // after restart. Here we just ignore this because we use a Memory storage.
+                _last_apply_index = entry.get_index();
+
+                if entry.get_data().is_empty() {
+                    // Emtpy entry, when the peer becomes Leader it will send an empty entry.
+                    continue;
+                }
+
+                if entry.get_entry_type() == EntryType::EntryNormal {
+                    // TODO: This usage of Raft only proposes one block at a time. Performance
+                    // could be improved by optimistically proposing blocks before parents are
+                    // committed.
+                    let block_id: BlockId = BlockId::from(Vec::from(entry.get_data()));
+                    debug!("Peer({}) committing block {:?}", self.id, block_id);
+                    if match self.leader_state {
+                        Some(LeaderState::Proposing(ref proposed)) => {
+                            &block_id == proposed
+                        },
+                        _ => false,
+                    } {
+                        debug!("Leader({}) transitioning to Committing block {:?}", self.id, block_id);
+                        self.leader_state = Some(LeaderState::Committing(block_id.clone()));
+                    }
+                    self.service.commit_block(block_id.clone()).expect("Failed to commit block");
+                }
+
+                // TODO: handle EntryConfChange
```

Comment:
Remove TODO.  If this is a remaining, unhandled case, a comment describing the case may be helpful